packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -194,31 +187,33 @@ export class DidExchangeProtocol {
     // Get DID Document either from message (if it is a supported did:peer) or resolve it externally
     const didDocument = await this.resolveDidDocument(agentContext, message)
 
-    if (isValidPeerDid(didDocument.id)) {
```

Comment:
Does this mean we now support connections with other dids than did:peer?

Reply:
I didn't do tests other than using the 'InMemory DID registry', but it should work, especially now we have the did caching (although for messaging maybe we should use a long-term cache instead of the default of 5 minutes).

Reply:
Okay makes sense.

Yes the 5 minutes was really to prevent multiple loads within a single exchange, without needing to worry about DIDs changing over time

Reply:
We could look at fancy caching strategies for this, like `stale-while-revalidate`: https://web.dev/articles/stale-while-revalidate